hnn_core/viz.py

```diff
@@ -1222,3 +1235,294 @@ def plot_laminar_csd(times, data, contact_labels, ax=None, colorbar=True,
     plt_show(show)
 
     return ax.get_figure()
+
+
+class NetworkPlotter:
+    """Helper class to visualize full
+       morphology of HNN model.
+
+    Parameters
+    ----------
+    net : Instance of Network object
+        The Network object
+    ax : instance of matplotlib Axes3D | None
+        An axis object from matplotlib. If None,
+        a new figure is created.
+    vmin : int | float
+        Lower limit of colormap for plotting voltage
+        Default: -100 mV
+    vmax : int | float
+        Upper limit of colormap for plotting voltage
+        Default: 50 mV
+    bg_color : str
+        Background color of ax. Default: 'black'
+    voltage_colormap : str
+        Colormap used for plotting voltages
+        Default: 'viridis'
+    elev : int | float
+        Elevation 3D plot viewpoint, default: 10
+    azim : int | float
+        Azimuth of 3D plot view point, default: 20
+    xlim : tuple of int | tuple of float
+        x limits of plot window. Default (-200, 3100)
+    ylim : tuple of int | tuple of float
+        y limits of plot window. Default (-200, 3100)
+    zlim : tuple of int | tuple of float
+        z limits of plot window. Default (-300, 2200)
+    trial_idx : int
+        Index of simulation trial plotted. Default: 0
+    time_idx : int
+        Index of time point plotted. Default: 0
+    """
+    def __init__(self, net, ax=None, vmin=-100, vmax=50, bg_color='black',
+                 voltage_colormap='viridis', elev=10, azim=-500,
+                 xlim=(-200, 3100), ylim=(-200, 3100), zlim=(-300, 2200),
+                 trial_idx=0, time_idx=0):
+        import matplotlib.pyplot as plt
+        from matplotlib import colormaps
+        self.net = net
+        self.times = net.cell_response.times
+
+        _validate_type(vmin, (int, float), 'vmin')
+        _validate_type(vmax, (int, float), 'vmax')
+        self._vmin = vmin
+        self._vmax = vmax
+
+        self._bg_color = bg_color
+        self._voltage_colormap = voltage_colormap
+
+        self.colormaps = colormaps  # Saved for voltage_colormap update method
+        self.colormap = colormaps[voltage_colormap]
+
+        # Axes limits and view positions
+        _validate_type(xlim, tuple, 'xlim')
+        _validate_type(ylim, tuple, 'ylim')
+        _validate_type(zlim, tuple, 'zlim')
+        _validate_type(elev, (int, float), 'elev')
+        _validate_type(azim, (int, float), 'azim')
+
+        self._xlim = xlim
+        self._ylim = ylim
+        self._zlim = zlim
+        self._elev = elev
+        self._azim = azim
+
+        # Trial and time indices
+        _validate_type(trial_idx, int, 'trial_idx')
+        _validate_type(time_idx, int, 'time_idx')
+
+        self._trial_idx = trial_idx
+        self._time_idx = time_idx
+
+        # Get voltage data and corresponding colors
+        self.vsec_array = self._get_voltages()
+        self.color_array = self.colormap(self.vsec_array)
```

Comment:
It'd be nice if we could populate `self.color_array` with zeros or ones at the onset so that the network can be plotted even if the simulation is run with `record_vsec=False`.

Reply:
Super good suggestion! Also makes for a good smoke test to plot a simulated/unsimulated network

Reply:
Overall, this branch is such a cool feature! Super excited to get it merged.